client/src/Molecules.elm

```diff
@@ -107,53 +117,43 @@ a Backbone with one or multiple Level 0 verctors.
 -}
 type alias Level1 =
     { name : String
+    , application : Maybe Application
```

Comment:
Do the different vector all need this? What is it used for?

Reply:
1. Sure, I actually have no idea where it comes from...

2. I was indeed playing with it with the idea to have the application moved to Level1 instead of the model because it makes more sense there. The problem is that the the overhangShape depends on the application chosen but since vectorToAdd is Nothing, it couldn't be used. I'll remove it from the Level1.

3. The vector isn't initialized until the first field is filled in. One way I could fix it is to initialize the vectorToAdd field when you call the SwitchPage msg. It would be something like this:

```elm
        SwitchPage page ->
            case page of
                ConstructLevel1 ->
                    ( { model
                        | page = page
                        , vectorToAdd = Just <| LevelNVec initLevel1
                      }
                    , Cmd.none
                    )

                AddLevel0Page ->
                    ( { model
                        | page = page
                        , vectorToAdd = Just <| Level0Vec initLevel0
                      }
                    , Cmd.none
                    )

                AddBackbonePage ->
                    ( { model
                        | page = page
                        , vectorToAdd = Just <| BackboneVec initBackbone
                      }
                    , Cmd.none
                    )

                _ ->
                    ( { model
                        | page = page
                        , vectorToAdd = Nothing
                      }
                    , Cmd.none
                    )
```

Reply:
> the overhangShape depends on the application

This sounds a lot like a data modelling issue.

> One way I could fix it is to initialize the vectorToAdd field when you call the SwitchPage msg

I strongly prefer you don't do this (at least not yet). Let's leave it for now